packages/ketcher-core/src/application/render/renderers/PolymerBondRenderer.ts

```diff
@@ -37,6 +53,231 @@ export class PolymerBondRenderer extends BaseRenderer {
   }
 
   public appendBond(rootElement) {
+    if (PolymerBondRenderer.isSnake) {
+      this.appendSnakeBond(rootElement);
+    } else {
+      this.appendBondGraph(rootElement);
+    }
+    return this.bodyElement;
+  }
+
+  public appendSnakeBond(rootElement) {
+    const startPosition = this.polymerBond.startPosition;
+    const endPosition = this.polymerBond.endPosition;
+    this.updateSnakeBondPath(startPosition, endPosition);
+
+    this.bodyElement = rootElement
+      .append('path')
+      .attr('stroke', this.polymerBond.finished ? '#333333' : '#0097A8')
+      .attr('stroke-width', 2)
+      .attr('class', 'selection-area')
+      .attr('d', this.path)
+      .attr('fill-opacity', 0)
+      .attr('pointer-events', 'stroke');
+    return this.bodyElement;
+  }
+
+  private getMonomerWidth() {
+    return this.polymerBond.firstMonomer.renderer?.bodyWidth ?? 0;
+  }
+
+  private getMonomerHeight() {
+    return this.polymerBond.firstMonomer.renderer?.bodyHeight ?? 0;
+  }
+
+  private updateSnakeBondPath(startPosition, endPosition) {
+    if (this.isSecondMonomerBottomRight(startPosition, endPosition)) {
```

Comment:
Seems this function is too long, could we extract different conditions code to different sub-functions?